.github/workflows/enos-run.yml

```diff
@@ -12,24 +12,28 @@ on:
 
 env:
   PKG_NAME: vault
+  ARTIFACT_NAME: ${{ inputs.artifact-name }}
 
 jobs:
+  # Read Enos scenario matrix file based on artifact-name input to test
+  read-enos-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      enos-scenarios: ${{ steps.enos-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Create Enos scenario matrix
+        id: enos-matrix
+        run: |
+          [[ ${{ env.ARTIFACT_NAME }} == *"ent"* ]] && echo "::set-output name=matrix::$(cat ./.github/scripts/ent.json)" || echo "::set-output name=matrix::$(cat ./.github/scripts/oss.json)"
+  # Run Integration tests on Enos scenario matrix
   enos:
     name: Integration
+    needs: read-enos-matrix
     strategy:
       fail-fast: false # don't fail as that can skip required cleanup steps for jobs
-      matrix:
-        # Run four scenarios to get a maximal distribution of variants in as
-        # few jobs as possible.
-        include:
-          - scenario: "smoke backend:consul consul_version:1.12.3 distro:ubuntu seal:awskms"
-            aws_region: "us-west-1"
-          - scenario: "smoke backend:raft consul_version:1.12.3 distro:ubuntu seal:shamir"
-            aws_region: "us-west-2"
-          - scenario: "upgrade backend:raft consul_version:1.11.7 distro:rhel seal:shamir"
-            aws_region: "us-west-1"
-          - scenario: "upgrade backend:consul consul_version:1.11.7 distro:rhel seal:awskms"
-            aws_region: "us-west-2"
+      matrix: ${{ fromJson(needs.read-enos-matrix.outputs.enos-scenarios) }}
```

Comment:
Confusing at first, now I get it. This is the GH Actions matrix. Do we want to do it this way, since very soon we will be able to run scenarios in parallel. That said, admittedly I don't know if GH Actions runs each matrix item in parallel, maybe it does.

Reply:
it does.  The matrix generated will be run in parallel